backbone/src/main/java/org/researchstack/backbone/result/logger/DataLoggerManager.java

```diff
@@ -0,0 +1,207 @@
+package org.researchstack.backbone.result.logger;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.annotation.MainThread;
+import android.util.Log;
+
+import com.google.gson.Gson;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Created by TheMDP on 2/7/17.
+ *
+ * The DataLoggerManager class is central station for the DataLogger file's life cycle
+ *
+ * It works by associating a "FileStatus" with each DataLogger file that is created
+ * The FileStatus can be "dirty", which means it is either being created or has not reached
+ * the stage of attempted upload.  Once the FileStatus is attempted upload, it will exist
+ * until it is uploaded.  After it is uploaded, the necessary parts of the file needed by
+ * the app can be stored, but the underlying data will be removed.
+ */
+
+public class DataLoggerManager {
+
+    /**
+     * SharedPreferences are used to keep track of the DataLogger files' status
+     * They can be dirty while being written, completed, and uploaded
+     */
+    private SharedPreferences sharedPrefs;
+    private static final String SHARED_PREFS_KEY = "DataLoggerManagerSharedPrefs";
```

Comment:
Interesting use of SharedPrefs here.  I think its fine.  I have done similar things by writing a second file to track status.  That could eliminate the dependency on the android context and allow you test more easily.